src/components/VerticalNavigation/VerticalNavigationItem.js

```diff
@@ -0,0 +1,236 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { ListGroup, ListGroupItem } from '../ListGroup';
+import OverlayTrigger from '../OverlayTrigger';
+import Tooltip from '../Tooltip';
+import VerticalNavigation from './VerticalNavigation';
+
+/**
+ * VerticalNavigationItem - a child element for the VerticalNavigation component
+ */
+export default class VerticalNavigationItem extends React.Component {
+  constructor() {
+    super();
+    this.state = {};
+  }
+
+  renderBadges(badges) {
+    const { showBadges } = this.props;
+    return (
+      showBadges &&
+      badges && (
+        <div className="badge-container-pf">
+          {badges.map(badge => (
+            <OverlayTrigger
+              key={badge.badgeClass}
+              placement="right"
+              overlay={<Tooltip>{badge.tooltip}</Tooltip>}
+            >
+              <div className={cx('badge', badge.badgeClass)}>
+                {badge.count &&
+                  badge.iconStyleClass && (
+                    <span className={badge.iconStyleClass} />
+                  )}
+                {badge.count && <span>{badge.count}</span>}
+              </div>
+            </OverlayTrigger>
+          ))}
+        </div>
+      )
+    );
+  }
+
+  render() {
+    const { item, ...otherProps } = this.props;
+    if (item) {
+      return <VerticalNavigationItem {...otherProps} {...item} />; // The item object is just a container for a bunch of props.
+    }
+
+    const {
+      children,
+      trackActiveState,
+      trackHoverState,
+      mobileItem,
+      iconStyleClass,
+      title,
+      badges,
+      showMobileSecondary,
+      showMobileTertiary,
+      navCollapsed,
+      handlePrimaryHover,
+      handleSecondaryHover,
+      handleTertiaryHover,
+      handlePrimaryBlur,
+      handleSecondaryBlur,
+      handleTertiaryBlur,
+      primaryItem,
+      secondaryItem,
+      tertiaryItem,
+      inMobileState,
+    } = this.props;
+
+    const handleHover = {
+      primary: handlePrimaryHover,
+      secondary: handleSecondaryHover,
```

Comment:
Re-reading this bit... I almost wonder if splitting out Secondary and Tertiary into their own components makes more sense (as it seems it would split the components a bit better as far as props/handlers/etc, i.e.:

```
<VerticalNavigation secondaryCollapsed={false} tertiaryCollapsed>
   <VerticalNavigation.Item title="Item 1">
      <VerticalNavigation.SecondaryItem title="Item 1-A"> 
         <VerticalNavigation.TertiaryItem title="Item 1-A-i">
      </VerticalNavigation.SecondaryItem> 
   <VerticalNavigation.Item>
</VerticalNavigation>         
```
It seems like this may split the logic a bit better and make it a bit easier to compose...
@jeff-phillips-18 thoughts?

Reply:
I may be over simplifying a bit though... not sure.